### DISCUSSION TITLE
🛠️ Project ideas to Build with Anon Aadhaar

### DISCUSSION THREAD

**Discussion by andyguzmaneth** [anon-aadhaar/anon-aadhaar#155] [Ideas]
### ⚙️ Protocol Improvements
- Explore different security methods for full lifecycle (expiration, revokation, etc)

### 🧩 Integrations 
-  Anon Aadhaar + Farcaster: gated channels and/or frames ([priv.cast](https://ethglobal.com/showcase/priv-cast-hzitq))
- GitcoinPassport-AnonAadhaar: create an integration to use Anon Aadhaar as a new element to prove your persoonhood on [Gitcoin Passport](https://docs.passport.gitcoin.co/building-with-passport/introduction)
- Anon Aadhaar + [Semaphore](https://github.com/semaphore-protocol): Build enhanced private voting apps.
### 📱 Apps Ideas
- **Aadhaar Wallet:** similar to [MynaWallet](https://ethglobal.com/showcase/myna-uxzdd), create an ERC-4337 compatible wallet that uses your Aadhaar card to approve transactions or social recover with other users.
- **Aadhaar Quadratic Funding / Voting:** vote if you can prove your citizenship. Implement a stack for Gitcoin, Optimism RPG, Giveth, or another big public good org using Quadratic Voting or Quadratic Funding. This will prepare for an Indian-focused grant rounds.
- **Anon Checkin (ETH India):** find a way to use Anon Aadhaar to anonymously, automatically, yet sybil-resistance, way to check-in to ETH India hackathons.
- **Anon Indian Stack:** i don’t know how, but add privacy too all possible layers in the indian stack.
- **UPI<>AnonAadhaar:** build a module that allows to claim an onchain payment anonymously with UPI. (explore Metamask UPI integration)
- **Micro-Loan Approval Platform:** create a platform for secure and anonymous loan approvals based on Aadhaar information. ******************Extra:****************** reputation is tracked anonymously using [UniRep Protocol](https://twitter.com/unirep_protocol?lang=en). **Extra 2:** identities are pseudonymous an integrates [[Semaphore](https://website.semaphore.pse.dev/)](https://website.semaphore.pse.dev/) for anonymous signaling.
- **Payment Channel:** use Anon Aadhaar SDK to create payment channel. Help people can verify another party with zkp. This is only for demo how people can use our SDK.
- [Bandada](https://pse.dev/projects/bandada) credential groups: gatekept by Anon Aadhaar proofs and then integrated to ****anonymous chats using [Discreetly](https://pse.dev/projects/discreetly).
- **Privacy-Preserving Defolio Community:** decentralized developer communities where members can engage in discussions and collaborations while maintaining their anonymity.
- **HeyIndia:** a copy of [HeyAnon](https://heyanon.xyz/) app, but need to prove you’re from India in order to post.
- **Telegram private groups:** where you need to prove you’re an Indian citizen in order to join
- **Web3 API Access Control:** develop a dApp that enables developers to control access to their web3 APIs based on verified Aadhaar identities.
- **SSO Server:** anonymously login with your “proof of citizenship” in any website. Explore integrations with Sign in with Ethereum
- **Artistic Social Commentary Platform**: Develop a platform where artists can post artworks that make social or political commentaries. Anon Aadhaar verification can ensure the authenticity of the contributors while protecting their identities, fostering a safe space for free expression.
- **Decentralized Art Critique Platform**: Create a platform where art critics and enthusiasts can provide feedback or critique on artworks. Anon Aadhaar would ensure that critiques are genuine and unbiased, focusing purely on the art rather than the critic’s reputation or relationship with the artist.
- **Local Feedback App**: An app where residents can anonymously or pseudonymous provide feedback or suggestions for local businesses and services. Anon Aadhaar would ensure that the feedback is from actual residents, helping businesses improve based on genuine community input.
- **Local Talent and Services Hub**: A platform where people can offer or seek local services (like gardening, carpentry, tutoring). Anon Aadhaar could verify the users & their reputation, ensuring safe and reliable exchanges within the community.
- **AnonAadhaar-UBI:** create a webapp that: i) allows individuals to fund a pool of [UBI](https://en.wikipedia.org/wiki/Universal_basic_income) by depositing onchain ii) Indians can claim their daily UBI by proving citizenship (and only once per day). ([AnonUBI](https://ethglobal.com/showcase/anonubi-bd8xg))

**Comment by burdges** [comment]
It's likely straightforward to make your circom groth16 circuts work on arkworks curves using  https://github.com/arkworks-rs/circom-compat/ 

At that point, a circom groth16 circuit could run fast on substrate, and evetually kusama & polkadot, using the upcoming EC host calls via  https://github.com/paritytech/arkworks-extensions

Just fyi, there is still some time left on this grants program: https://github.com/pioneersprize/Polkadot-Pioneers-Prize
